experimental/devices/epd/image2bit/image2bit.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package image2bit implements two bit gray scale (white, light gray,
+// dark gray, black) 2D graphics.
+//
+// It is compatible with package image/draw.
+//
+// The bit packing format is the same as used by waveshare e-Paper
+// displays such as the 4.2 inch display.
+package image2bit
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+)
+
+// Gray implements a 2 bit color.
+type Gray byte
+
+// RGBA returns either black, dark gray, light gray or white.
+func (b Gray) RGBA() (uint32, uint32, uint32, uint32) {
+	switch b {
+	case 0:
+		return 0, 0, 0, 65535
+	case 1:
+		return 0x5555, 0x5555, 0x5555, 0xffff
+	case 2:
+		return 0xaaaa, 0xaaaa, 0xaaaa, 0xffff
+	default:
+		return 0xffff, 0xffff, 0xffff, 0xffff
+	}
+}
+
+func (b Gray) String() string {
+	switch b {
+	case 0:
+		return "black"
+	case 1:
+		return "dark gray"
+	case 2:
+		return "light gray"
+	default:
+		return "white"
+	}
+}
+
+// All possible colors
+const (
+	White     Gray = 3
+	LigthGray Gray = 2
+	DarkGray  Gray = 1
+	Black     Gray = 0
+)
+
+// GrayModel is the color Model for 2 bit gray scale.
+var GrayModel = color.ModelFunc(convert)
+
+// BitPlane is a 2 bit gray scale image. To match the wire format
+// for waveshare e-Paper the two bits per pixel is stored across two bitmaps.
+// PixMSB contains the most significant bit, PixLSB contains the least significant bit.
+//
+//	        White LightGray DarkGray Black
+//	PixMSB   1     1         0        0
+//	PixLSB   1     0         1        0
+//
+// The following example shows the stored data for an 8 pixel wide image, 1 pixel high:
+// PixMSB []byte{0b10100000}
+// PixLSB []byte{0b10000000}
+//
+// It has a black background, the first pixel is white, and the third pixel LightGray.
+type BitPlane struct {
+	// PixMSB holds the image's most significant bit as a horizontally packed bitmap.
+	PixMSB []byte
+	// PixLSB holds the image's least significant bit as a horizontally packed bitmap.
+	PixLSB []byte
+	// Rect is the image's bounds.
+	Rect image.Rectangle
+}
+
+// NewBitPlane returns an initialized BitPlane instance, all black. TODO: Find a better name.
+func NewBitPlane(r image.Rectangle) *BitPlane {
+	// TODO: If any display has a width % 8 != 0, there might be
```

Comment:
At line 90, do  something like `((w+7)&^7) * h` so you don't need the TODO here. You'll want to update getOffset().

Reply:
I did not do it because I didn't have a display to test it on, but you are right, that is the most likely format.

I will check if I can rotate the display to 300x400 instead of 400x300, because 300%8 != 0 :)

Reply:
It only supports 400x300, 320x300, 320x240, 200x300 and flipping horizontally/vertically, so I can't test on this display.